ci/cray/Jenkinsfile.verbs

```diff
@@ -0,0 +1,140 @@
+
+pipeline {
+    options {
+        // Generic build options
+        timeout (time: 30, unit: 'MINUTES')
+        buildDiscarder(logRotator(numToKeepStr: '5'))
+
+        // Build options
+        disableConcurrentBuilds()
+        skipStagesAfterUnstable()
+        timestamps()
+    }
+    agent {
+        node {
+            label 'wham'
+        }
+    }
+    stages {
+        stage('Prepare') {
+            steps {
+                echo "creating git short hash"
+                script {
+                    GIT_SHORT_COMMIT = sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%h'").trim()
+                }
+                echo "checking for prerequisite test software"
+
+                echo " -- checking for presence of fabtests"
+                sh 'test -e $FABTEST_PATH/bin/runfabtests.sh'
+            }
+        }
+        stage('Build') {
+            steps {
+                sh './autogen.sh'
+                sh './configure --prefix=$TMP_INSTALL_PATH'
+                sh 'make -j 12'
+                sh 'make install'
+            }
+        }
+        stage('Test: Phase 1') {
+            failFast true
+            parallel {
+                stage('Unit tests') {
+                    steps {
+                        echo 'placeholder'
+                    }
+                }
+                stage('Smoke tests') {
+                    environment {
+                        LD_LIBRARY_PATH = "$TMP_INSTALL_PATH/lib:$LD_LIBRARY_PATH"
+                    }
+                    steps {
+                        echo 'checking for the presence of the verbs provider'
+                        sh 'srun -p mellanox -n 1 -N 1 $TMP_INSTALL_PATH/bin/fi_info -p \'ofi_rxm;verbs\''
+                    }
+                }
+                stage('Functional tests') {
+                    steps {
+                        echo 'placeholder'
+                    }
+                }
+                stage('Fabtests') {
+                    environment {
+                        LD_LIBRARY_PATH = "$TMP_INSTALL_PATH/lib:$LD_LIBRARY_PATH"
+                    }
+                    steps {
+                        // ignore the return code for fabtests until we can establish a fingerprint
+                        timeout (time: 20, unit: 'MINUTES') {
+                            sh 'srun -p mellanox -n 1 -N 1 -w wham-0-cn8 $FABTEST_PATH/bin/runfabtests.sh -p $FABTEST_PATH/bin -v -T 60 \'ofi_rxm;verbs\' 10.100.49.8 10.100.49.9 || true'
+                        }
+                    }
+                }
+            }
+        }
+        stage("Deploy-Install") {
+            steps {
+                sh "make clean"
+                sh "./configure --prefix=$ROOT_BUILD_PATH/libfabric/$GIT_SHORT_COMMIT"
+                sh "make -j 12"
+                sh "make install"
+            }
+        }
+        stage("Deploy-Development") {
+            when {
+                expression { env.BRANCH_NAME == 'master' }
+            }
```

Comment:
For development, should the branch name be not master?

Reply:
The thought here was that development would be something that passed smoke tests, but not necessarily MPI/systems tests. We don't have a development branch so, this is simply a point to 'latest' -- even if it might be unstable.